StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1129UnitTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Tunnel Vision Laboratories, LLC. All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+namespace StyleCop.Analyzers.Test.ReadabilityRules
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Analyzers.ReadabilityRules;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using TestHelper;
+    using Xunit;
+
+    /// <summary>
+    /// Unit tests for the <see cref="SA1129DoNotUseDefaultValueTypeConstructor"/> class.
+    /// </summary>
+    public class SA1129UnitTests : CodeFixVerifier
+    {
+        /// <summary>
+        /// Verifies that new expressions for reference types will not generate diagnostics.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task VerifyReferenceTypeCreationAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    public void TestMethod()
+    {
+        var v1 = new TestClass2();
+        var v2 = new TestClass2(1);
+    }
+
+    private class TestClass2
+    {
+        public TestClass2()
+        {
+        }
+
+        public TestClass2(int x)
+        {
+        }
+    }
+}
+";
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Verifies that new expressions for value types with parameters will not generate diagnostics.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task VerifyValueTypeWithParametersCreationAsync()
+        {
+            var testCode = @"public class TestClass
+{
+    public void TestMethod()
+    {
+        var v1 = new TestStruct(1);
+        var v2 = new TestStruct(1) { TestProperty = 2 };
+        var v3 = new TestStruct() { TestProperty = 2 };
```

Comment:
Nice catch finding this one!

:exclamation: You need to add one more for the following case (I tried adding it during the merge but it resulted in an AD0001 failure):

```csharp
var v4 = new TestStruct { TestProperty = 2 };
```